docs/directory-structure.md

```diff
@@ -0,0 +1,214 @@
+All directories in this file are relative to project root
+
+Each directory should start with three hashes 
+
+### .github
+configuration files that interact with GitHub services
+
+### .github/workflows
+workflow (interchangeably, actions) that runs by GitHub Actions service when code pushes to GitHub.
+
+### CreeDictionary
+
+A Django site with _django apps_ inside
+
+### CreeDictionary/API
+One of the django apps of our project. It defines click-in-text (our other project) API. 
+
+### CreeDictionary/API/migrations
+Django Migrations of the API app
+
+Migrations are not used to migrate database on production
+The only times migrations are needed are when we re-build the database and
+create empty tables with `manage.py migrate`
+
+This includes when we manually build test database, 
+and when our unit tests build test in memory database. 
+
+0000_initial.py migration is re-generated based on models.py everytime when the database is rebuilt.  
+
+This directory is managed by the rebuild script and does not require manual intervention
+
+### CreeDictionary/CreeDictionary
+One of the django apps of our project. It's the main part of our online dictionary.
+
+### CreeDictionary/CreeDictionary/static
+static files (A django concept) for the CreeDictionary app. They are usually javascripts and images
+
+
+### CreeDictionary/CreeDictionary/templates
+templates (A django concept, usually for HTML) for the CreeDictionary app.
+
+### CreeDictionary/CreeDictionary/static/CreeDictionary
+see below
+
+### CreeDictionary/CreeDictionary/templates/CreeDictionary
+You might think there is no stupid reason such verbose-looking directories should exist. But it's Django. 
```

Comment:
It took me a while to figure out the method to this madness. It makes sense now, but :S